rotors_gazebo_plugins/src/gazebo_fw_dynamics_plugin.cpp

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright 2017 Pavel Vechersky, ASL, ETH Zurich, Switzerland
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// MODULE HEADER
+#include "rotors_gazebo_plugins/gazebo_fw_dynamics_plugin.h"
+
+#include "ConnectRosToGazeboTopic.pb.h"
+
+namespace gazebo {
+
+GazeboFwDynamicsPlugin::GazeboFwDynamicsPlugin()
+    : ModelPlugin(),
+      node_handle_(0),
+      W_wind_speed_W_B_(0, 0, 0),
+      delta_aileron_left_(0.0),
+      delta_aileron_right_(0.0),
+      delta_elevator_(0.0),
+      delta_flap_(0.0),
+      delta_rudder_(0.0),
+      throttle_(0.0),
+      pubs_and_subs_created_(false) {
+}
+
+GazeboFwDynamicsPlugin::~GazeboFwDynamicsPlugin() {
+  event::Events::DisconnectWorldUpdateBegin(updateConnection_);
+}
+
+void GazeboFwDynamicsPlugin::Load(physics::ModelPtr _model,
+                                  sdf::ElementPtr _sdf) {
+  if (kPrintOnPluginLoad) {
+    gzdbg << __FUNCTION__ << "() called." << std::endl;
+  }
+
+  gzdbg << "_model = " << _model->GetName() << std::endl;
+
+  // Store the pointer to the model.
+  model_ = _model;
+  world_ = model_->GetWorld();
+
+  namespace_.clear();
+
+  // Get the robot namespace.
+  if (_sdf->HasElement("robotNamespace"))
+    namespace_ = _sdf->GetElement("robotNamespace")->Get<std::string>();
+  else
+    gzerr << "[gazebo_fw_dynamics_plugin] Please specify a robotNamespace.\n";
+
+  // Create the node handle.
+  node_handle_ = transport::NodePtr(new transport::Node());
+
+  // Initisalise with default namespace (typically /gazebo/default/).
```

Comment:
Oops. This is why they say not to copy-paste code :) ... I'll fix this in other plugins, too.